File: Programmers/Level 1/체육복.py
#Level 1 체육복
def solution(n, lost, reserve):
    delete = [] # lost와 resrve에서 중복으로 존재해서 삭제되는 학생들 저장
    for i in range(len(lost)) : 
        if lost[i] in reserve : 
            delete.append(lost[i])
            
    for i in range(len(delete)) : # 중복 학생 삭제
        lost.remove(delete[i])
        reserve.remove(delete[i])
    
    stu_num = n - len(lost) # 현재 수업을 들을 수 있는 학생 수
    
    for i in range(len(lost)) : 
        if lost[i]-1 in reserve : # 여벌을 빌릴 수 있으면 
            reserve.remove(lost[i]-1) # 여벌을 빌리고 reserve에서 빌려준 학생 삭제
            stu_num += 1 # 수업을 들을 수 있는 학생 수 추가
        elif lost[i]+1 in reserve : 
            reserve.remove(lost[i]+1)
            stu_num += 1
            
    answer = stu_num
    
    return answer

File: Programmers/Level 1/체육복.java
#Level 1 체육복(java)
import java.util.Arrays;
class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int[] delete = new int[0];
        for(int i=0;i<lost.length;i++){
            for(int j=0;j<reserve.length;j++){
                if(lost[i]==reserve[j]){
                    int add = delete.length;
                    delete = Arrays.copyOf(delete,add+1);
                    delete[add] = lost[i];
                    break;
                }
            }
        }
        for(int i=0;i<delete.length;i++){
            for(int j=0;j<lost.length;j++){
                if(delete[i]==lost[j]){
                    lost[j]=0;
                }
            }
            for(int j=0;j<reserve.length;j++){
                if(delete[i]==reserve[j]){
                    reserve[j]=0;
                }
            }
        }
        
        int lost_count = 0;
        for(int i=0;i<lost.length;i++){
            if(lost[i]!=0){
                lost_count++;
            }
        }
        int stu_num = n - lost_count;
        
        for(int i=0;i<lost.length;i++){
            for(int j=0;j<reserve.length;j++){
                if(lost[i] != 0 && reserve[j] !=0 && lost[i]-1 == reserve[j]){
                    lost[i] = 0;
                    reserve[j] = 0;
                    stu_num ++;
                }
                else if((lost[i] != 0) && reserve[j] !=0 && (lost[i]+1 == reserve[j])){
                    lost[i] = 0;
                    reserve[j] = 0;
                    stu_num ++;
                } 
            }
        }
        int answer = stu_num;
        return answer;
    }
}
